src/Parser/Mage/Fire/Modules/Items/MarqueeBindingsOfTheSunKing.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import ItemDamageDone from 'Main/ItemDamageDone';
+import ItemLink from 'common/ItemLink';
+import { formatPercentage } from 'common/format';
+
+class MarqueeBindingsOfTheSunKing extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  
+  damage = 0;
+  legendaryProcs = 0;
+  overwrittenProcs = 0;
+  buffedPyroblasts = 0;
+  expiredProcs = 0
+  
+  lastBuffTimestamp;
+  lastPyroCastTimestamp;
+  
+  on_initialized() {
+  this.active = this.combatants.selected.hasWrists(ITEMS.MARQUEE_BINDINGS_OF_THE_SUN_KING.id);
+  }
+
+	// get total count of procs
+  
+    on_toPlayer_applybuff(event) {
+	  if (event.ability.guid !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+	  return;
+    }
+  this.legendaryProcs += 1;
+  }
+  
+   // get count of refreshes
+  	on_byPlayer_refreshbuff(event) {
+    if (event.ability.guid !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+			return;
+		}
+		this.overwrittenProcs += 1;
+		this.legendaryProcs += 1;
+	}
+	
+	
+	// this section needs explaining:
+	// from looking through logs, the order of events is buff expires -> pyroblast cast begins within ~100ms -> pyroblast hits within 4.5 seconds
+	// i'm not sure this is the best way to accomplish this but it seems to work
+	
+	//get timestamp when buff expires
+	
+	on_byPlayer_removebuff(event) {
+    if (event.ability.guid !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+			return;
+		}
+		this.lastBuffTimestamp = this.owner.currentTimestamp;
+			}
+	
+	// for each pyroblast, check if buff expired within the past 100ms. if so, count it as buffed by the legendary
+	// additionally, set timestamp of cast to do a damage check
+	// i suppose it's possible for a player to instant cast pyroblast with hot streak within 100ms of the buff ending so it checks that this isn't the case.
+	
+	on_byPlayer_cast(event) {
+    if (event.ability.guid !== SPELLS.PYROBLAST.id) {
+			return;
+		}
+    if (this.lastBuffTimestamp > this.owner.currentTimestamp - 100 && !this.combatants.selected.hasBuff(SPELLS.HOT_STREAK) ) {
```

Comment:
the `hasBuff` function has an optional 'buffer' field that could be used instead of saving timestamps like this. Basically, this check could instead be `if (this.combatants.selected.hasBuff(SPELLS.KAELTHAS_ULTIMATE_ABILITY.id, event.timestamp, 100)) {`. See hasBuff in the Entity class for more details on the function.

Reply:
Huh, didn't know that - maybe I could refactor some of my Warlocks' code using this buffer, thanks!